sqlsynthgen/providers.py

```diff
@@ -34,3 +36,49 @@ class Meta:
     def bytes(self) -> bytes:
         """Return a UTF-8 encoded sentence."""
         return Text(self.locale).sentence().encode("utf-8")
+
+
+class TimedeltaProvider(BaseProvider):
+    """A Mimesis provider of timedeltas."""
+
+    class Meta:
+        """Meta-class for TimedeltaProvider settings."""
+
+        name = "timedelta_provider"
+
+    def timedelta(
+        self,
+        min_dt: Any = dt.timedelta(seconds=0),
+        # ints bigger than this cause trouble
+        max_dt: Any = dt.timedelta(seconds=2**32),
+    ) -> dt.timedelta:
+        """Return a random timedelta object."""
+        min_s = min_dt.total_seconds()
+        max_s = max_dt.total_seconds()
+        seconds = random.randint(min_s, max_s)
+        return dt.timedelta(seconds=seconds)
+
+
+class TimespanProvider(BaseProvider):
+    """A Mimesis provider for timespans, consisting of start datetime, end datetime, and
```

Comment:
Could you shorten the opening line of the docstring to <= 80 chars and move the rest to the body? Perhaps:

```suggestion
    """A Mimesis provider for timespans.
    
    A triple of start datetime, end datetime, and the timedelta in between.
    """
```

Reply:
Done. We generally stick to 88 chars, right? A special rule for first lines of docstrings to follow the PEP8 max of 79?

Reply:
Good question. It [looks like](https://peps.python.org/pep-0008/#maximum-line-length) the recommended line length for docstrings and comments is actually 72 chars, regardless of whether you increase the max allowed code line length. Though they don't really give a justification for that.